app/main.py

```diff
@@ -1,3 +1,55 @@
+from collections.abc import Callable
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Callable) -> Callable:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Callable) -> Callable:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        self.km = Distance(self.km + other)
+        return self
```

Comment:
I thought self.km should be a value, not a Distance instance

Reply:
my bad